src/listen/listener-registry.js

```diff
@@ -148,60 +157,51 @@ class ListenerRegistry {
 	 * @public
 	 * @returns {void}
 	 */
-	onSubscriptionMade( name, socketWrapper, count ) {
-		if( this.hasActiveProvider( name ) && this._topic === C.TOPIC.RECORD ) {
-			socketWrapper.send( messageBuilder.getMsg(
-				this._topic, C.ACTIONS.SUBSCRIPTION_HAS_PROVIDER, [ name, C.TYPES.TRUE ]
-			) );
+	onSubscriptionMade( subscriptionName, socketWrapper, localCount ) {
+		if( this.hasActiveProvider( subscriptionName ) ) {
+			this._listenerUtils.sendHasProviderUpdateToSingleSubscriber( true, socketWrapper, subscriptionName );
 			return;
 		}
 
-		if( count > 1 ) {
+		if( localCount > 1 ) {
 			return;
 		}
 
-		this._createListenMap( name );
-		this._triggerNextProvider( name );
+		this._startDiscoveryStage( subscriptionName );
 	}
 
 	/**
 	 * Called by the record subscription registry whenever the subscription count increments.
 	 * Part of the subscriptionListener interface.
 	 *
-	 * @param   {String} name
+	 * @param   {String} subscriptionName
 	 *
 	 * @public
 	 * @returns {void}
 	 */
-	onSubscriptionRemoved( subscriptionName, socketWrapper, count ) {
-		const provider = this._providedRecords[ subscriptionName ];
+	onSubscriptionRemoved( subscriptionName, socketWrapper, localCount, remoteCount ) {
+		const provider = this._locallyProvidedRecords[ subscriptionName ];
```

Comment:
Yeah. It's sole functionality is to send subscription removed, which means it needs a real client ( socket ) as opposed to cluster.

On subscription removed is called when both local and cluster subscriptions are changed.